gips/0065-subgraph-availability-manager.md

```diff
@@ -0,0 +1,123 @@
+---
+GIP: 0065
+Title: Subgraph Availability Manager
+Authors: Miguel de Elias <miguel@edgeandnode.com>
+Created: 2024-01-10
+Stage: Draft
+Discussions-To: https://forum.thegraph.com/t/gip-0065-subgraph-availability-manager/5501
+Category: Protocol Interfaces
+Implementations: https://github.com/graphprotocol/contracts/pull/882
+Audits: https://github.com/graphprotocol/contracts/blob/main/packages/contracts/audits/OpenZeppelin/2024-02-subgraph-availability-manager-and-minimum-allocation-duration-removal.pdf
+---
+
+# Abstract
+
+This GIP proposes increasing the decentralization in The Graph by introducing a new smart contract `SubgraphAvailabilityManager`.
+
+# Motivation
+
+The Subgraph Availability Oracle (SAO) plays a crucial role in The Graph's ecosystem. Its main function is to verify the availability and validity of subgraph files. Currently there is a single instance of the SAO in operation executing open-source code accessible at [The Graph's Subgraph Oracle repository](https://github.com/graphprotocol/subgraph-oracle). This SAO instance runs at five-minute intervals, performing a series of checks to assess the state of subgraph files. In instances where the SAO identifies unavailability or invalidity in a subgraph's files, it sends a transaction to the `RewardsManager` contract to deny indexing rewards for that subgraph.
+
+The proposed change aims to enhance the decentralization of the protocol by introducing five separate oracles operators and requiring a minimum of three votes among them before any decisive action is taken. All oracles operators will be bound by the **Subgraph Availability Oracle Operator Charter** included in this GIP which outlines their roles, responsibilities, and operational guidelines.
+
+# High-Level Description
+
+The new smart contract `SubgraphAvailabilityManager` will be exclusively deployed to the Arbitrum One network and will function as a mediator between five distinct oracle operators and the `RewardsManager` contract. It will feature a vote aggregation system and enforce an execution threshold requiring three votes to alter the state of a subgraph deployment within the `RewardsManager` contract. Additionally, it will incorporate a `voteTimeLimit` timeframe to monitor current votes, after which they expire and become invalid.
```

Comment:
It might be worth pointing out what is the default availability status for subgraphs (presumably "available"), and that no initial action or vote is typically necessary when a new subgraph is added to the network

Reply:
Sounds good, I can include that in `Motivation` together with the brief explanation of how the SAO works.